lib/Ziti.swift

```diff
@@ -456,9 +414,43 @@ import CZitiPrivate
             let refresh_interval = 30
         #endif
         
+        // convert kes and id info to char * types that ziti-sdk-c can use.
+        // also considered .withCString - https://stackoverflow.com/questions/31378120/convert-swift-string-into-cchar-pointer
+        let ctrlPtr = UnsafeMutablePointer<Int8>.allocate(capacity: id.ztAPI.count + 1)
+        ctrlPtr.initialize(from: id.ztAPI, count: id.ztAPI.count + 1)
+        
+        let certPEMPtr = UnsafeMutablePointer<Int8>.allocate(capacity: certPEM.count + 1)
+        certPEMPtr.initialize(from: certPEM, count: certPEM.count + 1)
+
+        let privKeyPEMPtr = UnsafeMutablePointer<Int8>.allocate(capacity: privKeyPEM.count + 1)
+        privKeyPEMPtr.initialize(from: privKeyPEM, count: privKeyPEM.count + 1)
+        
+        let caPEMPtr = UnsafeMutablePointer<Int8>.allocate(capacity: id.ca!.count + 1)
+        caPEMPtr.initialize(from: id.ca!, count: id.ca!.count + 1)
+        
+        // set up the ziti_config with our cert, etc.
+        var zitiCfg = ziti_config(
+            controller_url: ctrlPtr,
+            id: ziti_id_cfg(cert: certPEMPtr, key: privKeyPEMPtr, ca: caPEMPtr),
+            cfg_source: nil) // todo what is cfg_source?
+        
+        var zitiStatus = ziti_context_init(&self.zitiCtx, &zitiCfg)
+        guard zitiStatus == Ziti.ZITI_OK else {
+            let errStr = String(cString: ziti_errorstr(zitiStatus))
+            log.error("unable to initialize Ziti context, \(zitiStatus): \(errStr)", function:"start()")
+            initCallback(ZitiError(errStr, errorCode: Int(zitiStatus)))
+            return
+        }
+        
+        // todo will this happen automatically via gc?
```

Comment:
`allocate`/`deallocate` calls need to be paired, and the memory is not collected by Swift Automatic Reference Counting (ARC) when you explicitly allocate() it.  There are some docs on when to use `deinitialize()` that I'll try to dig up for you.